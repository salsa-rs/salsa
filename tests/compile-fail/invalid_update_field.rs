#[derive(salsa::Update)]
struct S2<'a> {
    bad2: &'a str,
}

fn main() {}
