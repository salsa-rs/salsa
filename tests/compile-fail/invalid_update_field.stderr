error: lifetime may not live long enough
 --> tests/compile-fail/invalid_update_field.rs:1:10
  |
1 | #[derive(salsa::Update)]
  |          ^^^^^^^^^^^^^ requires that `'a` must outlive `'static`
2 | struct S2<'a> {
  |           -- lifetime `'a` defined here
  |
  = note: this error originates in the derive macro `salsa::Update` (in Nightly builds, run with -Z macro-backtrace for more info)
